parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -254,8 +260,10 @@ EOF
         logs_to_events "AKS.CSE.ensureNoDupOnPromiscuBridge" ensureNoDupOnPromiscuBridge
     fi
 
-    if [ "$OS" = "$UBUNTU_OS_NAME" ] || isMarinerOrAzureLinux "$OS"; then
-        logs_to_events "AKS.CSE.ubuntuSnapshotUpdate" ensureSnapshotUpdate
+    if ! isAzureLinuxOSGuard "$OS" "$OS_VARIANT"; then
```

Comment:
why not ensure snapshot updates on OS guard?

Reply:
We dont support snapshot updates on OS Guard because the /usr partition is immutable, blocking package updates on running systems.